### ISSUE TITLE
next-auth not working with serverless-nextjs

### ISSUE THREAD

**Issue by jeremygottfried** [nextauthjs/next-auth#1481]
### Describe the bug
I deployed my app to Cloudfront, S3, and AWS Lambda Edge using `serverless-nextjs` component offered by serverless.com.

I am getting a 503 error with [next-auth](https://github.com/nextauthjs/next-auth) when I try to redirect to `/api/auth/signin`. I also get the same error if I try to load the signin page directly without a redirect.

> The Lambda function associated with the CloudFront distribution is invalid or doesn't have the required permissions. We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.

![Screen Shot 2021-03-08 at 3 56 48 PM](https://user-images.githubusercontent.com/34523280/110381353-57336d00-8027-11eb-9237-3928f1cb812b.png)

Code:
```js
export async function getServerSideProps(context) {
  const session = await getSession(context);

  if (!session) {
    return {
      redirect: {
        destination: '/api/auth/signin',
        permanent: false,
      },
    }
  }
...
}
```

Serverless.yml:
```yml
myNextApp:
  org: myOrg
  stage: prod
  component: "@sls-next/serverless-component@1.18.0" # it is recommended you pin the latest stable version of serverless-next.js
  inputs:
    bucketName: myApp-nextjs
    region: us-east-1
    name: myApp
```

### Actual behavior
Next-auth api endpoints are not working.

### Expected behavior
I expected the app to work the same way it does locally. I redirect to a signin page generated by next-auth when the user is not logged in.

**Comment by jeremygottfried** [comment]
503 error is gone after I upgraded to latest next-auth version, but it is failing to find prisma binaries within the lambda execution environment

**Comment by balazsorban44** [comment]
have you returned which Prisma adapter do you use? The built-in one, or the one here https://github.com/nextauthjs/adapters/tree/canary/packages/prisma

**Comment by jeremygottfried** [comment]
I believe I am using the built-in one:

```js
import Adapters from 'next-auth/adapters'

...
adapter: Adapters.Prisma.Adapter({ prisma })
```

> ERROR	[next-auth][error][get_user_by_provider_account_id_error] 
https://next-auth.js.org/errors#get_user_by_provider_account_id_error PrismaClientInitializationError2 [PrismaClientInitializationError]: Query engine binary for current platform "rhel-openssl-1.0.x" could not be found.
This probably happens, because you built Prisma Client on a different platform.
(Prisma Client looked in "/tmp/query-engine-rhel-openssl-1.0.x")

> Searched Locations:
> 
> 
> 
> You already added the platforms "native", "rhel-openssl-1.0.x" to the "generator" block
> in the "schema.prisma" file as described in https://pris.ly/d/client-generator,
> but something went wrong. That's suboptimal.
> 
> Please create an issue at https://github.com/prisma/prisma-client-js/issues/new
>     at PrismaClientFetcher.request (/var/task/pages/api/auth/[...nextauth].js:151931:15)
>     at processTicksAndRejections (internal/process/task_queues.js:97:5) {
>   clientVersion: '2.16.1'
> }

I have tried many different solutions, talked about here: 
- [serverless-next.js#843]( https://github.com/serverless-nextjs/serverless-next.js/issues/843)
- [prisma/docs#841](https://github.com/prisma/docs/issues/841)
- [next-auth#1191](https://github.com/nextauthjs/next-auth/issues/1191)
- [prisma/prisma#2195](https://github.com/prisma/prisma/issues/2195)
- [prisma-client-js#616](https://github.com/prisma/prisma-client-js/issues/616)
- [prisma/prisma#5392](https://github.com/prisma/prisma/issues/5392)

I'm going to reopen this issue in prisma and serverless-next.js because it seems to be more of an issue on their end, but if you have any ideas how to solve I would be very greatful!

**Comment by jeremygottfried** [comment]
Here's the issue I set up, in case anyone comes here looking for solutions: [https://github.com/serverless-nextjs/serverless-next.js/issues/937](https://github.com/serverless-nextjs/serverless-next.js/issues/937)

**Comment by cweekly** [comment]
TLDR it's an issue with Prisma: [prisma#6032](https://github.com/prisma/prisma/issues/6032).

**Comment by tranthaison1231** [comment]
I have same issue with next-auth and serverless-nextjs 

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/43759215/159427387-cddc57e1-d850-4535-ba63-2d494c5a3030.png">

Serverless.yml file: 
```
service: smartos-client-web
stage: ${env.STAGE}
region: ${env.REGION}
custom:
  baseName: ${service}-${stage}
web:
  component: '@sls-next/serverless-component@latest'
  inputs:
    # roleArn: ${env.ROLE}
    minifyHandlers: true
    logLambdaExecutionTimes: true
    useServerlessTraceTarget: true
    # build:
    #   postBuildCommands: ['npm run postbuild']
    memory:
      defaultLambda: 1024
      apiLambda: 1024
      imageLambda: 1024
    runtime:
      defaultLambda: 'nodejs14.x'
      apiLambda: 'nodejs14.x'
      imageLambda: 'nodejs14.x'
    timeout:
      defaultLambda: 20
      apiLambda: 15
      imageLambda: 15
    name:
      defaultLambda: ${custom.baseName}-default-lambda
      apiLambda: ${custom.baseName}-api-lambda
      imageLambda: ${custom.baseName}-image-lambda
    bucketName: ${custom.baseName}-bucket
    bucketRegion: ${region}
    cloudfront:
      distributionId: ${env.CLOUDFRONT_DISTRIBUTION_ID}
      defaults:
        forward:
          headers:
            [
              CloudFront-Is-Desktop-Viewer,
              CloudFront-Is-Mobile-Viewer,
              CloudFront-Is-Tablet-Viewer,
            ]
      api:
        minTTL: 10
        maxTTL: 10
        defaultTTL: 10
      aliases:
        - ${env.CLOUDFRONT_ALIAS}
      comment: ${env.CLOUDFRONT_ALIAS}
      priceClass: 'PriceClass_100'
      certificate:
        cloudFrontDefaultCertificate: false
        acmCertificateArn: ${env.CLOUDFRONT_ACM_CERTIFICATE_ARN}
        sslSupportMethod: 'sni-only'
        minimumProtocolVersion: 'TLSv1.2_2019'
      paths: ['/*']
      waitBeforeInvalidate: true

```

**Comment by tranthaison1231** [comment]
@ThangHuuVu Can you help me check this
